### PR TITLE
Add accessory bag overlay features

### PR DESCRIPTION
**Pull request by Madeleaan**
Currently includes:

- sort missing by MP
- highlight dupes
- make overall prettier
- in missing name + price
- on hover same as minions
- dont highlight missing deletors & compactors
- missing show only highest tiers

### CODE REVIEW

**Comment by NopoTheGamer** [comment]
opened it once, it said i was missing every talisman
opened it again, no missing talismans rendered https://hst.sh/raw/opowegakiv

File: src/main/java/io/github/moulberry/notenoughupdates/miscgui/AccessoryBagOverlay.java [hunk 1]
```diff
@@ -1059,4 +901,179 @@ public static int getRarity(ItemStack stack) {
...
+		if (rect.contains(mouseX(), mouseY())) {
+			String internal = NotEnoughUpdates.INSTANCE.manager
+				.createItemResolutionQuery()
+				.withItemStack(stack)
+				.resolveInternalName();
+			tooltipToDisplay = Arrays.asList(
+				stack.getDisplayName().replace("*", ""),
+				"",
+				"§eClick to learn more!",
+				"§eCtrl+Click to learn more!"
```
Comment (NopoTheGamer) [inline src/main/java/io/github/moulberry/notenoughupdates/miscgui/AccessoryBagOverlay.java:1046]: these shouldnt say the same thing

File: src/main/java/io/github/moulberry/notenoughupdates/miscgui/AccessoryBagOverlay.java [hunk 2]
```diff
@@ -1059,4 +901,179 @@ public static int getRarity(ItemStack stack) {
...
+		if (Mouse.isButtonDown(0) && internal != null) {
+			if (!Keyboard.isKeyDown(Keyboard.KEY_LCONTROL)) {
+				if (!NotEnoughUpdates.INSTANCE.manager.displayGuiItemRecipe(internal)) {
+					NEUOverlay.getTextField().setText("id:" + internal);
+					NotEnoughUpdates.INSTANCE.overlay.updateSearch();
+					NotEnoughUpdates.INSTANCE.overlay.setSearchBarFocus(true);
+				}
+			} else {
+				String displayname = Utils.cleanColour(stack.getDisplayName());
+				NotEnoughUpdates.INSTANCE.trySendCommand("/ahs " + displayname);
```
Comment (NopoTheGamer) [inline src/main/java/io/github/moulberry/notenoughupdates/miscgui/AccessoryBagOverlay.java:1062]: you need to get rid of the *

File: src/main/java/io/github/moulberry/notenoughupdates/miscgui/AccessoryBagOverlay.java [hunk 3]
```diff
@@ -1059,4 +901,179 @@ public static int getRarity(ItemStack stack) {
...
+			Minecraft.getMinecraft().currentScreen,
+			"guiLeft",
+			"field_147003_i"
+		);
+		int guiTop = (int) Utils.getField(
+			GuiContainer.class,
+			Minecraft.getMinecraft().currentScreen,
+			"guiTop",
+			"field_147009_r"
+		);
```
Comment (NopoTheGamer) [inline src/main/java/io/github/moulberry/notenoughupdates/miscgui/AccessoryBagOverlay.java:1007]:
```suggestion
		AccessorGuiContainer accessor = (AccessorGuiContainer) Minecraft.getMinecraft().currentScreen;
		int guiLeft = accessor.getGuiLeft();
		int guiTop = accessor.getGuiTop();
```

also in the other place its used

File: src/main/java/io/github/moulberry/notenoughupdates/miscgui/AccessoryBagOverlay.java [hunk 4]
```diff
@@ -1059,4 +901,179 @@ public static int getRarity(ItemStack stack) {
...
+			}
+		}
+
+		if (internal.contains("ABICASE")) {
+			try {
+				JsonObject profileInfo = GuiProfileViewer.getSelectedProfile().getProfileJson();
+				if (profileInfo.has("nether_island_player_data")) {
+					JsonObject data = profileInfo.get("nether_island_player_data").getAsJsonObject();
+					if (data.has("abiphone") && data.get("abiphone").getAsJsonObject().has("active_contacts")) { // BatChest
+						int contact = data.get("abiphone").getAsJsonObject().get("active_contacts").getAsJsonArray().size();
```
Comment (NopoTheGamer) [inline src/main/java/io/github/moulberry/notenoughupdates/miscgui/AccessoryBagOverlay.java:949]: also save this to config

File: src/main/java/io/github/moulberry/notenoughupdates/miscgui/AccessoryBagOverlay.java [hunk 5]
```diff
@@ -66,28 +73,42 @@
...
 @NEUAutoSubscribe
 public class AccessoryBagOverlay {
 	private static final int TAB_BASIC = 0;
 	private static final int TAB_TOTAL = 1;
-	private static final int TAB_BONUS = 2;
-	private static final int TAB_DUP = 3;
-	private static final int TAB_MISSING = 4;
-	private static final int TAB_OPTIMIZER = 5;
+	private static final int TAB_DUP = 2;
+	private static final int TAB_MISSING = 3;
```
Comment (NopoTheGamer) [inline src/main/java/io/github/moulberry/notenoughupdates/miscgui/AccessoryBagOverlay.java:85]: this could probably be an enum

**Review by NopoTheGamer — CHANGES_REQUESTED** [review]

**Comment by NopoTheGamer** [comment]
you could probably factor in the "current bonus" to the stats, it seems pretty standardized
image


also missing a few stats, might have to make a second page
image
image
image
image
Basically you could add all the stats from /equipment

File: src/main/java/io/github/moulberry/notenoughupdates/miscgui/AccessoryBagOverlay.java (hunk 1 above)
Comment (Madeleaan) [reply src/main/java/io/github/moulberry/notenoughupdates/miscgui/AccessoryBagOverlay.java:1046]: shit i forgot to change the string

**Comment by Madeleaan** [comment]
> opened it once, it said i was missing every talisman opened it again, no missing talismans rendered https://hst.sh/raw/opowegakiv

Sounds like it didnt load correctly the first time and you scrolled to a page that was beyond the last page it should show when it loads correctly, added an extra check

File: src/main/java/io/github/moulberry/notenoughupdates/miscgui/AccessoryBagOverlay.java [hunk 6]
```diff
@@ -1059,4 +901,179 @@ public static int getRarity(ItemStack stack) {
...
+					NotEnoughUpdates.INSTANCE.overlay.setSearchBarFocus(true);
+				}
+			} else {
+				String displayname = Utils.cleanColour(stack.getDisplayName());
+				NotEnoughUpdates.INSTANCE.trySendCommand("/ahs " + displayname);
+			}
+		}
+	}
+
+	public static boolean hasConsuedRiftPrism() {
```
[the remark this answers is not in the document]
Comment (Madeleaan) [reply src/main/java/io/github/moulberry/notenoughupdates/miscgui/AccessoryBagOverlay.java]: Hopefully did it right

**Review by NopoTheGamer — APPROVED** [review]

File: src/main/resources/assets/notenoughupdates/accessory_bag_overlay_v2.png
Comment (nea89o) [inline src/main/resources/assets/notenoughupdates/accessory_bag_overlay_v2.png:1]: image
Maybe this tab could blend in better.

**Review by nea89o — APPROVED** [review]

**Pull request merged**